drivers/SmartThings/zigbee-fan/src/fan-light/init.lua

```diff
@@ -0,0 +1,161 @@
+-- Copyright 2024 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local stDevice = require "st.device"
+local clusters = require "st.zigbee.zcl.clusters"
+local capabilities = require "st.capabilities"
+local FanControl = clusters.FanControl
+local Level = clusters.Level
+local OnOff = clusters.OnOff
+
+local FINGERPRINTS = {
+  { mfr = "Samsung Electronics", model = "SAMSUNG-ITM-Z-003" },
+}
+
+local function can_handle_itm_fanlight(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local levels_for_speed = {
+  [0] = 0,
+  [1] = 25,
+  [2] = 50,
+  [3] = 100,
+}
+
+local function level_to_speed(level)
+  local speed = level
+  if speed ~= nil then
+    if level == 0 then
+      speed  = 0
+    elseif level  > 0 and level <= 25 then
+      speed = 1
+    elseif level > 25 and level <= 75 then
+      speed = 2
+    else
+      speed = 3
+    end
+  else
+    speed = 0
+  end
+  return speed
+end
+
+-- CAPABILITY HANDLERS
+
+local function on_handler(driver, device, command)
+  local dev = device
+  if command.component == 'light' then
+    dev:send(OnOff.server.commands.On(dev))
+  else
+    local speed = dev:get_field('LAST_FAN_SPD') or 1
+    dev:send(FanControl.attributes.FanMode:write(dev,speed))
+    dev:send(FanControl.attributes.FanMode:read(dev))
+  end
+  dev:send(FanControl.attributes.FanMode:read(dev))
```

Comment:
you're reading `FanMode` twice in a row if the fan is turned on

Reply:
I will change it.